test/test_helper.rb

```diff
@@ -60,17 +60,13 @@ module MiniTest
   class Test
     module Help
       class << self
-        def binary(str)
-          str.force_encoding(Encoding::BINARY)
-        end
-
         def cache_path(dir, file, args_key = nil)
           hash = fnv1a_64(file)
           unless args_key.nil?
             hash ^= fnv1a_64(args_key)
           end
 
-          hex = hash.to_s(16)
+          hex = hash.to_s(16).rjust(16, '0')
```

Comment:
This is the main fix. `to_s(16)` If the first byte was 0, the returned string wouldn't be 16 char long, which would mess with the cache key generation.